src/python/zquantum/core/measurement.py

```diff
@@ -392,3 +393,115 @@ def convert_bitstring_to_int(bitstring: Iterable[int]) -> int:
             significant (little endian).
     """
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
+
+class Measurements:
+    """ A class representing measurements from a quantum circuit. """
+
+    def __init__(self, filename=None):
+        self.counts = {}
+        self.measurements = []
+        self.num_measurements = 0
+
+        if filename is not None:
+            with open(filename, "r") as f:
+                data = json.load(f)
+                self.add_measurements(data["measurements"])
+
+
+    def save(self, filename):
+        """ Serialize the Measurements object into a file in JSON format.
+        
+        Args:
+            filename (string): filename to save the data to 
+        """
+        data = { "schema":         SCHEMA_VERSION+"-measurements",
+                 "counts":         self.get_counts(),
+                 "measurements":   self.get_measurements()}
+        with open(filename, "w") as f:
+            f.write(json.dumps(data, indent=2))
+
+
+    def get_measurements(self):
```

Comment:
Why do you need a getter for that?
The fields in the method are not private, so you can just do:
`measurements.measurements`, not need for `measurements.get_measurements()`.
You now have two ways to do the same thing.

Reply:
Addressed in refactor here:
231a59a